src/requests/consumption.js

```diff
@@ -420,7 +426,636 @@ async function convertRecommendedRolesForProjects(recommendedFor) {
+				isAPrivateProgram: false,
+				allowMultipleAssessemts: resource?.allowMultipleAssessemts ? true : false,
+				isDeleted: false,
+				pageHeading: 'Domains',
+				minNoOfSubmissionsRequired: resource?.minNoOfSubmissionsRequired
+					? resource?.minNoOfSubmissionsRequired
+					: 1,
+				rootOrganisations: resource?.organization
+					? resource?.organization.map((organization) => organization.id)
+					: [],
+				createdFor: resource?.organization ? resource?.organization.map((organization) => organization.id) : [],
+				deleted: false,
+				name: resource?.title,
+				programExternalId: programDetails.externalId,
+				entityType: resource?.entityType ? resource?.entityType : null,
+				type: common.SOLUTIONS_TYPE[resource.type] ? common.SOLUTIONS_TYPE[resource.type] : null,
+				subType: common.SOLUTIONS_TYPE[resource.type] ? common.SOLUTIONS_TYPE[resource.type] : null,
+				isReusable: false,
+				externalId: utils.generateUniqueId(),
+				programId: programDetails._id,
+				programName: programDetails.name,
+				programDescription: programDetails.description,
+				status: common.STATUS_ACTIVE.toLowerCase(),
+				updatedAt: new Date(),
+				createdAt: new Date(),
+				scope: programDetails.scope,
+				projectTemplateId: resource._id,
+				updatedBy: 1,
+				endDate: programDetails.end_date,
+				startDate: programDetails.start_date,
+			}
+
+			solutionRolloutMap[solutionTemplate.externalId] = resource.rolloutId
+
+			// map resource externalId and certificate Data if it has certificate data
+			if (
+				resource?.certificate &&
+				typeof resource?.certificate === 'object' &&
+				Object.keys(resource?.certificate).length != 0
+			) {
+				solutionCertificateMap.push({
+					externalId: solutionTemplate.externalId,
+					certificate: resource?.certificate,
+				})
+			}
+			solutionsToCreate.push(solutionTemplate)
+		})
+
+		const solutionCollection = mongoDb.collection(COLLECTIONS.SOLUTIONS)
+		await solutionCollection.insertMany(solutionsToCreate)
+
+		const createdSolutions = await solutionCollection
+			.find({
+				programId: programDetails._id,
+			})
+			.toArray()
+
+		if (solutionCertificateMap && solutionCertificateMap.length > 0) {
+			solutionCertificateMap.forEach((solutionMap) => {
+				const found = createdSolutions.find((solution) => solution.externalId == solutionMap.externalId)
+				insertCertificateTemplate(found.certificate, found._id, programDetails._id)
+			})
+		}
+
+		return createdSolutions.map((solution) => {
+			return {
+				...solution,
+				rolloutId: solutionRolloutMap[solution.externalId],
+			}
+		})
+	} catch (error) {
+		console.log(error)
+	}
+}
+
+const duplicateResources = async (resourceDetails) => {
+	let projectTemplateIds = []
+	let solutionTemplateIds = []
+
+	if (resourceDetails.type == common.PROJECT) projectTemplateIds.push(ObjectId(resourceDetails._id))
+	else solutionTemplateIds.push(ObjectId(resourceDetails._id))
```

Comment:
In case of projects we have to create templates in ```projectTemplates``` collection. in case of any other solutions , template will be created in ```solutions``` collection. So based on type here it is being segregated. Logic to create template also changes based on type. So in future we can enhance this code. @priyanka-TL